guide/01-getting-started/package-json.md

```diff
@@ -119,3 +119,31 @@ import { someUtility } from 'my-logger/utils.js'; // This will throw an error
 When `exports` is provided, usually there's no need to specify the `main` field unless you need to support older versions of Node.js or some legacy tools that do not recognize the `exports` field.
 
 To learn more about the `exports` field, see [Node.js documentation on package exports](https://nodejs.org/api/packages.html#package-entry-points).
+
+## `package.json` lookups
+
+In Node.js, when a module is loaded using either `require()`, `import` or `import()`, Node.js will search for a `package.json` file starting from the directory in which the module being loaded resides. If it doesn't find one, it will continue searching up the directory tree until it either finds a `package.json` file or reaches the root of the filesystem. If any `package.json` file is found during this search, Node.js will use the fields in that file to determine how to load the module.
+
+This means that, for example, if you have a project structure like this:
+
+```
+my-project/
+├── node_modules/
+│   └── my-logger/
+│       ├── logger.js
+│       └── package.json
+├── src/
+│   └── app.js
+├── lib/
+│   ├── package.json
+│   └── utils.js
+└── package.json
+```
+
+Assuming `src/app.js` loads `../lib/utils.js`, which in turn loads `'my-logger'`, when you run `node src/app.js` from `my-project`:
+
+1. The `src/app.js` file is loaded with configuration from the top-level `package.json` file in `my-project`.
+2. The `../lib/utils.js` file is loaded with configuration from `lib/package.json`.
+3. The `my-logger` module is loaded using configuration from `node_modules/my-logger/package.json`.
+
+This means that different types of modules can coexist in the same project even if they all use the `.js` extension. For example if the `lib/package.json` file sets `"type": "commonjs"`, then `utils.js` can be written as a CommonJS module, even if the top-level `package.json` under `my-project` sets `"type": "module"` which allows `src/app.js` to be written as an ESM module. This is not recommended for new projects, but it can be useful when gradually migrating a codebase from CommonJS to ESM.
```

Comment:
I was going through the book and I think we should either remove this
https://github.com/nodejs/package-examples/blob/1b03a2ab2741de1dd9857a83b98c03b50e32169a/guide/01-getting-started/commonjs-and-esm.md?plain=1#L159-L161

 or add it here at the end of the package-json.md, because it looks like we are talking about ESM and CommonJS files but the next chapter is about package.json